### PR TITLE
Set ScyllaCluster placement requirements on cleanup Jobs

### LINKED ISSUE #1490: Cleanup jobs cannot be scheduled on tainted nodes
**Describe the bug**

When all nodes in cluster are tainted, cleanup jobs cannot run anywhere due to missing toleration.

**To Reproduce**
1. Taint all nodes in cluster, specify toleration for some taint in ScyllaCluster object
2. Create ScyllaCluster object
3. Cleanup Jobs are created but stuck in Pending state forever.

**Expected behavior**
Cleanup Jobs run on nodes tolerated by ScyllaCluster.

### PR DESCRIPTION
**Pull request by zimnx**
Missing placement might cause cleanup Jobs to be stuck in Pending state forever, preventing ScyllaCluster from fully boostrapping. Placement requirements are copied into cleanup Jobs from ScyllaCluster spec.

Resolves #1490

### CODE REVIEW

File: pkg/controller/scyllacluster/resource_test.go [hunk 1]
```diff
@@ -1876,6 +1876,194 @@ func TestMakeJobs(t *testing.T) {
 			},
 			expectedConditions: nil,
 		},
+		{
+			name: "cleanup job has the same placement requirements as ScyllaCluster",
+			cluster: func() *scyllav1.ScyllaCluster {
+				cluster := basicScyllaCluster.DeepCopy()
+				cluster.Spec.Datacenter.Racks[0].Placement = &scyllav1.PlacementSpec{
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource_test.go:1883]: should we have 2 racks to make sure we assign the correct, corresponding placement options?

Comment (zimnx) [reply pkg/controller/scyllacluster/resource_test.go:1883]: Good idea, added.

File: pkg/controller/scyllacluster/resource_test.go [hunk 2]
```diff
@@ -1876,6 +1876,380 @@ func TestMakeJobs(t *testing.T) {
...
+								},
+							},
+							Tolerations: []corev1.Toleration{
+								{
+									Key:      "role",
+									Operator: corev1.TolerationOpEqual,
+									Value:    "scylla-clusters",
+									Effect:   corev1.TaintEffectNoSchedule,
+								},
+							},
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource_test.go:1988]: please make the different from rack[0] as well

Comment (zimnx) [reply pkg/controller/scyllacluster/resource_test.go]: done

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm

thanks!

**Pull request merged**